code/modules/mob/living/carbon/human/species/station/skrell/skrell.dm

```diff
@@ -61,8 +61,39 @@
 
 	zombie_type = "Skrell Zombie"
 
+/datum/species/skrell/handle_environment_special(var/mob/living/carbon/human/H)
+	var/old_moisture = H.ST.moisture
+	H.ST.moisture = max(0, H.ST.moisture - rand(-0.8, 1.3))			// Reduce moisture a little.
+	if(H.fire_stacks && H.ST.moisture > 1800)						// Flaming moist skrell can reduce their flame
+		H.fire_stacks--
+		H.ST.moisture -= rand(150, 350)								// Lost a lot of moisture
+	var/pain_limit = (1800-H.ST.moisture) * (40 / 1800)				// The max halloss damage to do based on current moisture. At 0, this is 30. At 1800 or higher, this is 0.
+	
+	// Add more pain damage if they need it.
+	var setPain = max(H.getHalLoss(), pain_limit)
+	H.adjustHalLoss(setPain - H.getHalLoss())
```

Comment:
So if I'm reading your pain_limit comments right, at zero moisture adjustHalLoss would do 30.  handle_environment_special is called on every handle_environment tick; this would add 30 Halloss on every one of those ticks if they're dry, which pretty much guarantees they'll go into shock. Are you looking to set HalLoss instead?

Reply:
I couldn't find a setHalLoss proc, so I just adjustHalLoss relative to the current getHalLoss, hence setPain - H.getHalLoss

Reply:
Through investigations, the setHalLoss proc for humans just deals brute damage, which is... not ideal...

Reply:
You do not NEED to set it. You can make it so it happens over time, that's fine. But you'll probably want to make sure it doesn't happen frequently OR that it's not so much damage. 30 Halloss every environment tick will pretty much put them into pain-based shock and cardiac arrest in under a minute.

Reply:
It's not pain damage over time, it's a minimum amount of pain felt. Setting is required.

Reply:
To futher clarify, Skrell shouldn't ever be KO'd by a lack of water, just disadvantaged.